include/MacportsLegacySupport.h

```diff
@@ -124,10 +124,11 @@
 /* STAILQ_FOREACH is not defined on Tiger*/
 #define __MP_LEGACY_SUPPORT_STAILQ_FOREACH__  (__APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050)
 
-/* c++11 <cmath> PPC 10.[45] and Intel 10.[4-6], GNU g++ 4.6 through 8. */
-#if (__APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070 \
-               && defined(__GNUC__) && (__GNUC__ <= 8)                 \
-               && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 6))))
+/* c++11 <cmath> PPC 10.[45] and Intel 10.[4-6], GNU g++ 4.6 through 8, or for any clang. */
+#if (__APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1070  \
+     && ( defined(__clang__)                                            \
+          || defined(__GNUC__) && (__GNUC__ <= 8)                       \
```

Comment:
I would prefer to do `|| ( defined(__GNUC__) && ... && ... )` to make it clear that all of these clauses belong to the same compiler ...even though it's technically not necessary to do so

Reply:
@michaelld I've tried to re-format condition to make it clear and easy to review.

Reply:
To be honest I have no idea how to reformat it to make it easy to read :(

Reply:
LOL I can't think of a good way either! This latest is good enough IMHO ... thanks!